File: backend/src/main/java/tko/model/dto/nutrition/ProductDTO.java
package tko.model.dto.nutrition;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ProductDTO {
    private Long id;
    @NotBlank
    private String name;
    @NotNull
    private Integer kcal;
    @NotBlank
    private String unit;

    private LocalDateTime createdAt;

    private Integer likeCount;
    @NotNull
    private BigDecimal fats;
    @NotNull
    private BigDecimal carbs;
    @NotNull
    private BigDecimal proteins;
}

File: app/src/main/java/tk/ssau/fitnesstko/model/dto/nutrition/ProductDTO.kt
package tk.ssau.fitnesstko.model.dto.nutrition

import org.threeten.bp.LocalDateTime
import java.math.BigDecimal

data class ProductDTO(
    val id: Long?,
    val name: String,
    val kcal: Int,
    val unit: String,
    val grams: BigDecimal,
    val portion: BigDecimal,
    val createdAt: String,
    val fats: BigDecimal,
    val carbs: BigDecimal,
    val proteins: BigDecimal
)
